internal/process/defunct_processes.go

```diff
@@ -0,0 +1,89 @@
+package process
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// ProcessFS is the process file system.
+const ProcessFS = "/proc"
+
+// Stat represents status information of a process from /proc/[pid]/stat.
+type Stat struct {
+	// Comm is the command name (usually the executable filename).
+	Comm string
+
+	// State is the state of the process.
+	State string
+}
+
+// DefunctProcesses returns the number of zombie processes in the node.
+func DefunctProcesses() (defunctCount uint, retErr error) {
+	directories, err := os.Open(ProcessFS)
+	if err != nil {
+		logrus.Warn(err)
```

Comment:
nit: Adding descriptive errors with logrus.Warnf("opening proc filesystem failed: %v", err)... Probably should update the other logrus.Warns in the file as well.

Reply:
Thanks for the suggestion @rphillips! But I think `os.Open` returns enough context as @kolyshkin suggested in https://github.com/cri-o/cri-o/pull/5082#discussion_r672636392 so we may not need it :)